HLTrigger/HLTcore/src/TriggerExpressionL1uGTReader.cc

```diff
@@ -90,7 +95,41 @@ namespace triggerExpression {
       }
     }
 
+    m_triggersAfterMasking = m_triggers;
     m_initialised = true;
   }
 
+  void L1uGTReader::mask(Evaluator const& eval) {
+    auto const& triggersToMask = eval.triggers();
+
+    if (triggersToMask.empty()) {
+      edm::LogInfo("NoTriggersToMask") << "\tL1uGTReader[\"" << *this << "\"]::mask(arg = " << eval << ")"
+                                       << " failed: arg.triggers() is empty";
+      return;
+    }
+
+    // patterns() is always empty for a L1uGTReader, and not empty for PathReader;
+    // here, PathReader evaluators are skipped as they shouldn't be used to mask a L1uGTReader
+    if (not eval.patterns().empty()) {
+      edm::LogWarning("InvalidArgumentForMasking") << "\tL1uGTReader[\"" << *this << "\"]::mask(arg = " << eval << ")"
+                                                   << " failed: arg.patterns() is not empty (arg is not a L1uGTReader)";
+      return;
+    }
```

Comment:
This check and `return` (here, and in `PathReader`) is not strictly necessary, as L1T and HLT `triggers()` will never overlap in name by construction (the `find` call below would never give a match). Still, in such a case, it might be better to give a warning (as in the current implementation).